presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewPlanValidator.java

```diff
@@ -32,15 +41,36 @@
 {
     private final Node query;
 
+    private final boolean analyzeColumnMapping;
+    private final Optional<Analysis> analysis;
+
+    private ImmutableList.Builder<List<TableColumn>> linkedBaseColumns;
+
     public MaterializedViewPlanValidator(Node query)
     {
+        this(query, true, Optional.empty());
```

Comment:
analyzeColumnMapping has to be additional functionality, so it should be false by default.